packages/starlight/utils/navigation.ts

```diff
@@ -135,7 +135,7 @@ function makeLink(
 	attrs?: LinkHTMLAttributes
 ): Link {
 	if (!isAbsolute(href)) href = pathWithBase(href);
-	const isCurrent = href === ensureTrailingSlash(currentPathname);
+	const isCurrent = new URL(href, 'http://example.com').pathname === ensureTrailingSlash(currentPathname);
```

Comment:
I think [`encodeURI()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURI) should work in this case (I haven't tested it though yet) but if it does, it may be easier to understand than using a fake base URL? Any thoughts?

```suggestion
	const isCurrent = encodeURI(href) === ensureTrailingSlash(currentPathname);
```